providers/sendinblue/src/lib/sendinblue.provider.spec.ts

```diff
@@ -58,6 +61,35 @@ test('should trigger sendinblue library correctly', async () => {
   });
 });
 
+test('should correctly use sender email and name from the config', async () => {
+  const emailsApi = new TransactionalEmailsApi('/test');
+  const sendTransacEmailMock = jest.fn(
+    async (_sendSmtpEmail, _options?) => ({})
+  );
+  emailsApi.sendTransacEmail = sendTransacEmailMock as any;
+  const provider = new SendinblueEmailProvider(mockConfig);
+  // eslint-disable-next-line @typescript-eslint/dot-notation
+  provider['transactionalEmailsApi'] = emailsApi;
+  const { from, ...mockNovuMessageWithoutFrom } = mockNovuMessage;
+
+  // use config.from if message.from is not provided
+  await provider.sendMessage(mockNovuMessageWithoutFrom);
+  expect(sendTransacEmailMock).toBeCalledTimes(1);
+  expect(sendTransacEmailMock.mock.calls[0][0]).toBeInstanceOf(SendSmtpEmail);
+  const smptEmail = sendTransacEmailMock.mock.calls[0][0] as SendSmtpEmail;
+  expect(smptEmail.sender.email).toBe(mockConfig.from);
+
+  // sendinblue sender.name should be set from config
+  expect(smptEmail.sender.name).toBe(mockConfig.senderName);
+
+  // Use the message.from instead of config.from if avaliable
+  await provider.sendMessage(mockNovuMessage);
+  expect(sendTransacEmailMock).toBeCalledTimes(2);
+  const smptEmailWithFrom = sendTransacEmailMock.mock
+    .calls[1][0] as SendSmtpEmail;
+  expect(smptEmailWithFrom.sender.email).toBe(mockNovuMessage.from);
```

Comment:
I would suggest for readability to split these 3 cases in 3 different individual tests. Also that would help to isolate potential bugs.

Reply:
I feel this is already pretty readable and if there is a regression, it will be very easy to tell why based on which comments + failure line number. It is <100 lines along with informative comments.

Breaking it up into multiple tests just seems like extra work and adds boilerplate that I think harms readability 🤷🏽‍♂️

Reply:
I am sorry but fully disagree. If there is a regression you need to review all the content of this test and the 3 different flows. Having 3 separated tests, inside the CI integrations will quickly highlight which flow is the wrong one. 
For me the current readability has a lot of room to improve. For me, a test is readable if I read the test name and understand what's going to happen. And in this case I read the title and find one flow is going to be tested, to realise in the code that 3 different flows are being tested.

Reply:
**TLDR:** Thanks for your reply. I am still not convinced because there was no logical justification for improved readability. Your comments are not specific or actionable enough for me. I found your terminology vague. Seems like you're expecting me to apply some sort of "1 test, 1 flow" rule which seems like overkill for a 1-line change. I don't think there is a magic number of "flows" that make a test readable. Blocking the PR for this reason alone seems dosn't make sense to me, but I will defer to your judgement. Happy to keep discussing but if there is no real, programming-related changes to make, I will leave it here.

> If there is a regression you need to review all the content of this test and the 3 different flows. 

What is a "flow", exactly? Do you mean "code path" or "function call"? Are you saying we must only test 1 code path or function per test? Also, how can you fix a regression without reviewing the content of the failed test? Does not make sense.

> For me, a test is readable if I read the test name and understand what's going to happen. 

The title of the test is `should correctly use sender email and name from the config`. Which words / line numbers specifically are unclear to you? What is the unexpected part of the test / name for you?

> For me the current readability has a lot of room to improve. 

You claim this is because:

- you want only 1 flow per test (which you have not clearly defined)
- you want to somehow be able to tell why a test failed without looking at the test contents

Both these "rules" sound simplistic, dogmatic and quite frankly impossible. I don't even think the existing tests obey these rules. You have stated your own personal expectations around testing but fail to present any logic as to **_why_** your approach is better. 

I'm not a testing expert but I know that tests must be:

- **Well-scoped:** a test must have an appropriate scope (limit functionality being tested, integration vs unit, how mocks are applied etc.). Note that this has absolutely **nothing** to do with the "flows", assertions, function calls or # of lines. You must apply your common sense and judgement as you write individual tests.
- **Descriptively named:** the description of the test should accurately describe exactly what it does
- **Reliable:** tests should not be flakey, run reliably and consistently each time
- **Representative:** tests should mirror behavior in production as closely as possible

I claim that my tests already meet this common-sense criteria. I believe your requirement that "each test should only have 1 flow" is a gross oversimplification of a much more complex guideline. This is our main problem right now.

To be clear, this code is already **working** and **tested**. There are no application-layer bugs or logic problems we disagree about, only test readability. The change is low-risk. In these cases, I think it's always better to get working code deployed and iterate quickly rather than have people argue about their opinions on readability. [Something](https://medium.com/young-coder/dont-be-dogmatic-49029822d61d) to ponder... 

Happy to hear any clarifications in case I'm misunderstanding you. Nonetheless, if I was a maintainer I don't think I would nitpick _readability_ on a _test_ written by a _first-time contributor and potential customer_ to verify a minor **1-line fix**....seems like overkill for "readability".

If you still think '1 flow 1 test' rule will prevent this PR from being merged we can just close it...I guess at that point I'll just have to (sadly) say thank you for your wonderful OSS software and think about running it privately 

Again, happy to talk through any clarification if I've missed something.

![image](https://user-images.githubusercontent.com/1176261/204197747-06cbadfd-bb6a-46b9-8980-314f8aab0aca.png)

Reply:
Sorry if my explanation was vague and the terminology confusing. Also I made a mistake of reading 3 different potential tests inside your test, when now I can see there is only potentially 2. Apologies for that also.

This part of your test, does a test that what it does is something in the lines "should send an email using the `from` value from the `config` when missing `from` in the `message`":
```
 // use config.from if message.from is not provided
  await provider.sendMessage(mockNovuMessageWithoutFrom);
  expect(sendTransacEmailMock).toBeCalledTimes(1);
  expect(sendTransacEmailMock.mock.calls[0][0]).toBeInstanceOf(SendSmtpEmail);
  const smptEmail = sendTransacEmailMock.mock.calls[0][0] as SendSmtpEmail;
  expect(smptEmail.sender.email).toBe(mockConfig.from);

  // sendinblue sender.name should be set from config
  expect(smptEmail.sender.name).toBe(mockConfig.senderName);
  ```
  
This other part of your test, does a test that what it does is something in the lines "should send an email using the `from` value from `message`":
  ```
  // Use the message.from instead of config.from if avaliable
  await provider.sendMessage(mockNovuMessage);
  expect(sendTransacEmailMock).toBeCalledTimes(2);
  const smptEmailWithFrom = sendTransacEmailMock.mock
    .calls[1][0] as SendSmtpEmail;
  expect(smptEmailWithFrom.sender.email).toBe(mockNovuMessage.from);
  ```

That's why I mentioned _flows_ with the meaning of being 2 different use cases. Being both in the same test when they are mutually exclusive use cases when a notification is triggered is what leads me to think they should live in separated tests. Amongst other test implementation details.

To finalise, the name of your test reads: "should correctly use sender email and name from the config" that defines perfectly the first bit code I mention, but not the second bit, therefore hides a test needed for one of the use cases. So I would call to apply the bullet points `Descriptively named` and `Representative`, that you brilliantly wrote as desired test criteria, as reasons also for my request of changes.
Hope this clarifies better the reason of my petition and insistence. 👍🏻